src/main/java/com/exasol/adapter/installer/Installer.java

```diff
@@ -58,9 +56,11 @@
      * Install Postgres Virtual Schema to the Exasol database.
      */
     public void install() throws SQLException, BucketAccessException, TimeoutException, FileNotFoundException {
-        uploadFilesToBucket();
+        final Jar virtualSchemaJar = this.virtualSchemaJarProvider.provideJar();
+        final Jar jdbcDriverJar = this.jdbcDriverJarProvider.provideJar();
+        uploadFilesToBucket(virtualSchemaJar, jdbcDriverJar);
         try (final Connection connection = getConnection(); final Statement statement = connection.createStatement()) {
-            installVirtualSchema(statement);
+            installVirtualSchema(statement, List.of(virtualSchemaJar.getJarName(), jdbcDriverJar.getJarName()));
```

Comment:
I find it a bit cumbersome that we pass the Statement object as parameter through the whole flow. I wonder if we can do it differently, maybe extracting to an instance variable.
Also I wonder if reusing the statement is OK.

Reply:
I moved the statement creation one level lower. As we only execute 4 queries, I think it's fine to use one statement. The methods are basically string building + executing the statement